cylc/flow/scheduler.py

```diff
@@ -1262,11 +1262,17 @@ def _configure_contact(self) -> None:
 
     def load_flow_file(self, is_reload=False):
         """Load, and log the workflow definition."""
-        # Local workflow environment set therein.
-        # Allow -S and -D to take effect in Cylc VR.
-        # https://github.com/cylc/cylc-flow/issues/5968
-        self.options.rose_template_vars = []
-        self.options.defines = []
+        if is_reload:
+            # If the workflow is reloaded clear any existing rose options
+            # The reload command doesn't have the ability to set these but
+            # if the user has used VR to re-install before reload they will
+            # expect the changed values not the ones stored on the scheduler.
+            # Note: Does NOT apply to reload on restart, because the play
+            # command used to restart can have template variables attached.
+            # See https://github.com/cylc/cylc-flow/pull/5996
```

Comment:
This explanation doesn't make much sense to me as the play command used to start the workflow in the first place can also have template variables attached?

Reply:
If you do 

```
cylc play x --pause -S 'foo=1'
cylc stop x
cylc play x -S 'foo=2'
```
Do you want foo=1 or 2?